routes/apiRouter.js

```diff
@@ -19,208 +19,197 @@ routes.get('/', (req, res) => res.json({ message: 'Welcome to Imperial Bin\'s AP
 routes.post(['/document', '/postCode', '/paste'], (req, res) => {
     db.link.loadDatabase();
     const code = req.body.code;
-    if (req.headers.authorization || req.body.apiToken) {
-        const apiToken = req.headers.authorization || req.body.apiToken;
-        Users.findOne({ apiToken }, (err, user) => {
-            if (user) {
-                const longerUrls = req.body.longerUrls || false
-                const imageEmbed = req.body.imageEmbed || false
-                var expiration = req.body.expiration || 5
-                const instantDelete = req.body.instantDelete || false
-                const creator = user._id.toString()
-                var quality = 73;
-                var str = Math.random().toString(36).substring(2)
-                if (user.memberPlus)
-                    var quality = 100;
-                if (longerUrls)
-                    var str = Math.random().toString(36).slice(2) + Math.random().toString(36).slice(2) + Math.random().toString(36).slice(2);
-                if (expiration >= 31)
-                    var expiration = 31;
-                createPaste(str, imageEmbed, instantDelete, expiration, creator, quality);
-            } else {
-                createPaste(Math.random().toString(36).substring(2), false, false, 5, 'NONE');
-            }
-        })
-    } else {
-        createPaste(Math.random().toString(36).substring(2), false, false, 5, 'NONE');
-    }
+    // Anon function to quickly make a guest paste.
+    const guestPaste = () => { createPaste(Math.random().toString(36).substring(2), false, false, 5, 'NONE'); }
+
+    if(!req.headers.authorization || !req.body.apiToken) return guestPaste();
+    const apiToken = req.headers.authorization || req.body.apiToken;
+    
+    Users.findOne({ apiToken }, (err, user) => {
+        if(err) return throwApiError(res, "Sorry! There was a internal server error, please contact a administrator!");
+        if(!user) return guestPaste();
+
+        const creator = user._id.toString();
+        const userSettings = {
```

Comment:
could we rename this to documentSettings please?

Reply:
Done.